src/coreclr/src/tools/Common/TypeSystem/Ecma/EcmaField.cs

```diff
@@ -277,21 +279,65 @@ public override MarshalAsDescriptor GetMarshalAsDescriptor()
 
     public static class EcmaFieldExtensions
     {
+        /// <summary>
+        /// Returns the RVA associated with an RVA mapped field from the PE module.
+        /// </summary>
+        public static int GetFieldRvaValue(this EcmaField field)
+        {
+            Debug.Assert(field.HasRva);
+            return field.MetadataReader.GetFieldDefinition(field.Handle).GetRelativeVirtualAddress();
+        }
+
         /// <summary>
         /// Retrieves the data associated with an RVA mapped field from the PE module.
         /// </summary>
-        public static byte[] GetFieldRvaData(this EcmaField field)
+        public static unsafe byte[] GetFieldRvaData(this EcmaField field, int targetPointerSize)
         {
             Debug.Assert(field.HasRva);
-            int addr = field.MetadataReader.GetFieldDefinition(field.Handle).GetRelativeVirtualAddress();
-            var memBlock = field.Module.PEReader.GetSectionData(addr).GetContent();
+
+            MetadataReader metadataReader = field.MetadataReader;
+            int rva = metadataReader.GetFieldDefinition(field.Handle).GetRelativeVirtualAddress();
+            ImmutableArray<byte> memBlock = field.Module.PEReader.GetSectionData(rva).GetContent();
 
             int size = field.FieldType.GetElementSize().AsInt;
             if (size > memBlock.Length)
                 throw new BadImageFormatException();
 
-            byte[] result = new byte[size];
-            memBlock.CopyTo(0, result, 0, result.Length);
+            byte[] result = new byte[AlignmentHelper.AlignUp(size, targetPointerSize)];
```

Comment:
Could you place this new logic into CopiedFieldRvaNode? This is quite specific to what that node wants to do. It's not general purpose type system logic. The commonly expected definition of field rva data for a single field is what this method originally returned. R2R compiler doesn't have to use this method.

Reply:
Nothing else in the compiler uses this other than the node that write the data.  Should we just move all of it to CopyFieldRvaNode and remove these general purpose extension methods? What do you think?

Reply:
The general purpose extension method is still used on the CoreRT side. I expect we will eventually use it here too again e.g. if we ever add static data preinitialization a la .NET Native (getting the byte[] data in this original sense would be needed in expanding RuntimeHelpers.InitializeArray).

Reply:
Alright, then I’ll keep the existing logic the way it is and move the new stuff as you suggested.